src/org/abymaly/code/BinaryTree.java

```diff
@@ -0,0 +1,57 @@
+package org.abymaly.code;
+
+public class BinaryTree<T> {
+
+    public class Node<T> {
+        public Integer data;
+        public Node nextLeft;
+        public Node nextRight;
+
+        public Node(Integer data) {
+            this.data = data;
+            }
+
+        public Integer getData() {
+            return data;
+        }
+
+        public Node getNextLeft() {
+            return nextLeft;
+        }
+
+        public Node getNextRight() {
+            return nextRight;
+        }
+
+        Node<T> root;
+        Node<T> actual;
```

Comment:
Jaki jest powód, że lokalna zmienna metody append trafiła tu jako pole klasy???
Przecież to jest chwilowa zmienna konkretnej metody i dla klasy ona nic nie wnosi.